astropy/io/ascii/sextractor.py

```diff
@@ -64,6 +67,15 @@ def get_cols(self, lines):
                     column_unit = columns[previous_column][2]
                     columns[c] = (column_name, column_descr, column_unit)
             previous_column = n
+        if dataline is not None and len(colnumbers) > 0:  # handle the case where the last column is array-like
```

Comment:
Sorry that I was not able to work on this until today. I completely agree with you, so I made some changes to the code. It is now much cleaner.